backend/app/routers/users.py

```diff
@@ -0,0 +1,71 @@
+from fastapi import APIRouter, Depends, HTTPException, logger, status
+
+from app.authentication import get_current_user
+from app.models.rail_user import RailUser
+from app.schemas.rail_user import RailUserResponse
+
+router = APIRouter()
+
+
+@router.get("/users/profile")
+async def get_user_profile(
+    user: dict = Depends(get_current_user(required=True)),
+) -> RailUserResponse:
+    try:
+        user_obj = await RailUser.find_one({"email": user["email"]})
+
+        if not user_obj:
+            user_obj = await RailUser(email=user["email"]).create()
+    except Exception:
+        logger.error("Error while fetching or creating user profile")
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="Error while fetching or creating user profile",
+        )
+
+    return user_obj.map_to_response()
+
+
+@router.get("/users/api_key")
+async def get_user_api_key(
+    user: dict = Depends(get_current_user(required=False)),
+) -> str:
+    user = await RailUser.find_one({"email": user["email"]})
+
+    if not user or "api_key" not in user:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail="API key not found",
+        )
+    return user["api_key"]
+
+
+@router.post("/users/api_key")
+async def create_or_change_user_api_key(
+    user: dict = Depends(get_current_user(required=True)),
+) -> str:
+    user_obj = await RailUser.find_one({"email": user["email"]})
+
+    if not user_obj:
+        try:
+            user_obj = await RailUser(
+                email=user["email"], api_key=RailUser.generate_api_key()
+            ).create()
+        except Exception:
+            logger.error("Error while creating user profile and API key")
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail="Error while creating user profile and API key",
+            )
+    elif "api_key" not in user_obj or not user_obj.api_key:
```

Comment:
I don't believe there's any case when `api_key` attribute wouldn't be defined for the `user_obj`, so you only need to check whether `api_key` is empty or not. And I'm not sure whether the condition `"api_key" not in user_obj` is correct since the user_obj is an instance of a class, not a dictionary

Reply:
It is, when a user visits his/her profile for the first time. Then, only the profile is created, not the API key.